frontend/packages/schema-model/src/lib/SchemaModel.ts

```diff
@@ -69,17 +69,42 @@ export class SchemaModel {
   }
 
   public getRootNode(): FieldNode | CombinationNode {
-    const rootNode = this.getNode(ROOT_POINTER);
+    const rootNode = this.getNodeBySchemaPointer(ROOT_POINTER);
     if (!isFieldOrCombination(rootNode))
       throw new Error('Root node is not a field nor a combination.');
     return rootNode;
   }
 
-  public getNode(pointer: string): UiSchemaNode {
+  public getNodeBySchemaPointer(pointer: string): UiSchemaNode {
     if (!this.hasNode(pointer)) throw new Error(`Node with pointer ${pointer} not found.`);
     return this.nodeMap.get(pointer);
   }
 
+  public getNodeByUniquePointer(uniquePointer: string): UiSchemaNode {
+    const schemaPointer = this.getSchemaPointerByUniquePointer(uniquePointer);
+    return this.getNodeBySchemaPointer(schemaPointer);
+  }
+
+  public getSchemaPointerByUniquePointer(uniquePointer: string): string {
+    if (this.hasNode(uniquePointer)) return uniquePointer;
+
+    const definitionPointer = this.extractDefinitionByUniquePointer(uniquePointer);
+    const remainderPointer = uniquePointer.split('/').slice(3).join('/');
+    return `${definitionPointer}/${remainderPointer}`;
+  }
+
+  private extractDefinitionByUniquePointer(uniquePointer: string): string {
+    const parentPointer = uniquePointer.split('/').slice(0, 3).join('/');
+    const firstChildPointer = this.getChildNodes(parentPointer)[0].pointer;
+    return firstChildPointer.split('/').slice(0, 3).join('/');
+  }
+
+  public getUniquePointer(pointer: string, uniqueParentPointer?: string): string {
+    if (!uniqueParentPointer || !isDefinitionPointer(pointer)) return pointer;
+
+    return `${uniqueParentPointer}/properties/${extractNameFromPointer(pointer)}`;
+  }
```

Comment:
Dette ble mye bedre! Jeg synes likevel det er litt vanskelig å se hva som skjer her. Hvorfor er akkurat `firstChildPointer` av interesse? Vil dette fungere hvis man vil ha element nummer to? (Du må gjerne utvide testene med dette scenariet.)

Jeg har prøvd meg litt frem selv, og ser at flytskjemaet jeg lagde var litt mangelfullt. Her er i hvert fall en kode som fungerer:
```suggestion
  public getSchemaPointerByUniquePointer(uniquePointer: string): string {
    if (this.hasNode(uniquePointer)) return uniquePointer;
    const parentNodePointer = this.getParentSchemaPointerByUniquePointer(uniquePointer);
    return makePointerFromArray([parentNodePointer, Keyword.Properties, extractNameFromPointer(uniquePointer)]);
  }

  private getParentSchemaPointerByUniquePointer(uniquePointer: string): string {
    const parentPropertyNode = this.getParentPropertyNodeByUniquePointer(uniquePointer);
    return isReference(parentPropertyNode) ? parentPropertyNode.reference : parentPropertyNode.pointer;
  }

  private getParentPropertyNodeByUniquePointer(uniquePointer: string): UiSchemaNode {
    const parentUniquePointer = uniquePointer.split('/').slice(0, -2).join('/');
    return this.getNodeByUniquePointer(parentUniquePointer);
  }
```